pkg/server/postgres/postgres.go

```diff
@@ -39,13 +39,32 @@ func (p *PostgresPlugin) DatabasePlugin() plugin.DatabasePlugin {
 	return p
 }
 
-func (p *PostgresPlugin) Dsn(pw string, s *atlas.DatabaseServer) string {
-	return fmt.Sprintf("%s:%s@%s.%s:%d/postgres?sslmode=disable",
-		s.Spec.SuperUser,
-		pw,
-		s.Name,
-		s.Namespace,
-		s.Spec.Port)
+func (p *PostgresPlugin) Dsn(user string, pass string, db *atlas.Database, s *atlas.DatabaseServer) string {
+	if user == "" {
+		user = s.Spec.SuperUser
+	}
+	if pass == "" {
+		pass = s.Spec.SuperUserPassword
+	}
+
+	dbHost := fmt.Sprintf("%s.%s", s.Name, s.Namespace)
+	if s.Spec.Host != "" {
+		dbHost = s.Spec.Host
+	}
+	// TODO remove  this
+	dbHost = "192.168.39.216"
+
+	database := "postgres"
+	if db != nil {
+		database = db.Name
+	}
```

Comment:
mention why are you doing this. As in, when db can be nill

Reply:
ok